### PR TITLE
feat(gains): add gains plugin with options task support

### PR DESCRIPTION
**Pull request by rontero83**

### CODE REVIEW

File: packages/gains/src/Gains.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { Chains, OrderType } from '@rabbitholegg/questdk-plugin-utils'
+import { type Address } from 'viem'
+
+export const options = async (
+  params: OptionsActionParams,
+): Promise<TransactionFilter> => {
+  const { recipient, amount, orderType, chainId, contractAddress, token } =
+    params
+
+  if (chainId !== Chains.ARBITRUM_ONE) {
```
Comment (mmackz) [inline packages/gains/src/Gains.ts:26]: We want to retain the ability to add new chain in the future. If the chainId is not correct, the filter will not pass. We do not need to force failure.

**Review by Quazia — CHANGES_REQUESTED** [review]
Could you add some additional test coverage and make sure that the variables and filter are setup in a way that will be more compatible with adding more trading pairs and collateral contracts in the future?

File: packages/gains/src/Gains.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      _type: typeFilter,
+    },
+  })
+}
+
+function failingFilter(): Promise<TransactionFilter> {
+  return compressJson({
+    chainId: 9999999999,
+  })
+}
```
[the remark this answers is not in the document]
Comment (rontero83) [reply packages/gains/src/Gains.ts]: change to throw pushing. is it ok?

File: packages/gains/src/test-transactions.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    'when open trade and contractAddress not defined',
+    { contractAddress: undefined },
+  ),
+  createTestCase(OPTIONS_TEST, 'when open trade and orderType not defined', {
+    orderType: undefined,
+  }),
+  createTestCase(OPTIONS_TEST, 'when open trade and amount not defined', {
+    amount: undefined,
+  }),
+]
```
[the remark this answers is not in the document]
Comment (rontero83) [reply packages/gains/src/test-transactions.ts]: add WBTC test transaction

File: packages/gains/src/constants.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { type Address, zeroAddress } from 'viem'
+
+// https://arbiscan.io/address/0x2c7e82641f03fa077f88833213210a86027f15dc
+export const CONTRACT_ADDRESS: Address =
+  '0x2c7e82641f03fa077f88833213210a86027f15dc'
```
[the remark this answers is not in the document]
Comment (rontero83) [reply packages/gains/src/constants.ts:5]: renamed

**Comment by rontero83** [comment]
> Could you add some additional test coverage and make sure that the variables and filter are setup in a way that will be more compatible with adding more trading pairs and collateral contracts in the future?

Hi, @Quazia, @mmackz 
Have we any issues regarding to gain options pull request or not?

**Pull request closed without merging**